src/HttpOverStream.Server.Owin/CustomListenerHost.cs

```diff
@@ -179,6 +188,99 @@ public void Initialize(IAppBuilder builder)
             }
         }
 
-    }
+        private class WriteInterceptStream : Stream
+        {
+            private readonly Stream _innerStream;
+            private Func<Task> _once;

```

Comment:
Whats this for? Can we name it something to describe what it does (I realise you want to imply its only triggered once)

Reply:
Yes, it is an async func that should be called only once (the first time any variant of Write/Flush is called)

Reply:
What about something like 
`_actionBeforeFirstWrite`
`_beforeFirstWrite`
`_onWriteOnce`